js/sri4node-attachments.js

```diff
@@ -1,1078 +1,1677 @@
-
-              let filenames = sriRequest.attachmentsRcvd.filter(e => e.tmpFileName).map(e => e.tmpFileName);
-
-              if (filenames.length) {
-                try {
-                  await deleteFromS3(filenames);
-                  sriRequest.logDebug(filenames.join(" & ") + " deleted");
-                } catch (err) {
-                  sriRequest.logDebug("delete rollback failed");
-                  sriRequest.logDebug(err);
-                }
-              }
-
-              if (securityError) throw securityError;
-
-              throw failed;
-              //stream.push(failed);
-            } else {
-              /// all went well, rename the files to their real names now.
-              bodyJson.filter(e => e.file !== undefined).forEach(file => {
-                renames.push(
-                  renameFile(file)
-                );
-
-              });
-
-              await Promise.all(renames);
-
-              let response = [];
-              bodyJson.forEach(file => {
-                response.push({ status: 200, href: file.resource.href + "/attachments/" + file.attachment.key });
-              });
-              return response;
-            }
-          }
+          /// all files are now uploaded into their TMP versions.
+        } catch (err) {
+          await handleErrorDuringUploadOrCopy(err, sriRequest, copiedAttachmentsWithFileObj);
         }
-      },
 
-      customRouteForPreSignedUpload: function () {
+        /// all went well, rename the files to their real names now.
+        await renameAttachmentsToRealNames(copiedAttachmentsWithFileObj);
+
         return {
-          routePostfix: '/attachments/presigned',
-          httpMethods: ['GET'],
-          readOnly: true,
-          beforeHandler: async (tx, sriRequest) => {
-            // await checkSecurity(tx, sriRequest, null, 'create');
-          },
-          handler: async (tx, sriRequest) => {
-            ///dp the presigned request to s3
-            let json = await getPreSigned();
-            return {
-              body: json,
-              status: 200
-            }
-          }
+          status: 200,
+          body: copiedAttachmentsWithFileObj.map((file) => ({
+            status: 200,
+            href: `${file.resource.href}/attachments/${file.attachment.key}`,
+          })),
         };
       },
+    };
+  }
 
-
-      customRouteForDownload: function () {
+  /**
+   * A function that will generate a json object that can be used in
+   * sriConfig.resources.*.customRoutes in order to add a GET
+   * /resource/attachments/prsegignedupload route.
+   * This is something specific to upload a file directly to S3 or something
+   * (without proxying through the api server) by requesting some kind of token first
+   * and then using that to be allowed to upload something directly to S3.
+   *
+   * [ Currently not used ]
+   *
+   * @returns {TCustomRoute}
+   */
+  function customRouteForPreSignedUpload() {
```

Comment:
@stefanvds1 I see this route is commented out in the activityplans-api, do we still need it?